toshi_hazard_store/scripts/ths_ds_defrag.py

```diff
@@ -58,54 +56,64 @@ def write_metadata(base_path, visited_file):
 @click.command()
 @click.argument('source')
 @click.argument('target')
-@click.option("-p", "--parts", help="comma-separated list of partition keys for the target DS", default="")
+@click.option("-p", "--parts", help="comma-separated list of partition keys for the target DS", default="vs30,nloc_0")
 @click.option('-v', '--verbose', is_flag=True, default=False)
 def main(
     source,
     target,
     parts,
     verbose,
 ):
-    """Compact the dataset
+    """Compact and repartition the dataset.
 
     Can be used on both realisation and aggregate datasets.
-    """
-    source_folder = pathlib.Path(source)
-    target_folder = pathlib.Path(target)
-    target_parent = target_folder.parent
 
-    assert source_folder.exists(), f'source {source_folder} is not found'
-    assert source_folder.is_dir(), f'source {source_folder} is not a directory'
+    Arguments:\n
+
+    SOURCE: path to the source (folder OR S3 URI).\n
+    TARGET: path to the target (folder OR S3 URI).
+    """
 
-    assert target_parent.exists(), f'folder {target_parent} is not found'
-    assert target_parent.is_dir(), f'folder {target_parent} is not a directory'
+    source_dir, source_filesystem = pyarrow_dataset.configure_output(source)
```

Comment:
there is no longer an explicit check that the source and targets exist and are directories if they are filepaths. Is this handled implicit;y by pyarrow?

Reply:
I think there's enough low level checking to cover this